integrations/watsonx/src/haystack_integrations/components/embedders/watsonx/text_embedder.py

```diff
@@ -0,0 +1,171 @@
+# SPDX-FileCopyrightText: 2025-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Any
+
+from haystack import component, default_from_dict, default_to_dict
+from haystack.utils import Secret, deserialize_secrets_inplace
+from ibm_watsonx_ai import Credentials
+from ibm_watsonx_ai.foundation_models import Embeddings
+
+
+@component
+class WatsonXTextEmbedder:
+    """
+    Embeds strings using IBM watsonx.ai foundation models.
+
+    You can use it to embed user query and send it to an embedding Retriever.
+
+    ### Usage example
+
+    ```python
+    from haystack_integrations.components.embedders.watsonx.text_embedder import (
+        WatsonXTextEmbedder,
+    )
+
+    text_to_embed = "I love pizza!"
+
+    text_embedder = WatsonXTextEmbedder(
+        model="ibm/slate-30m-english-rtrvr",
+        api_key=Secret.from_env_var("WATSONX_API_KEY"),
+        url="https://us-south.ml.cloud.ibm.com",
+        project_id=Secret.from_env_var("WATSONX_PROJECT_ID"),
+    )
+
+    print(text_embedder.run(text_to_embed))
+
+    # {'embedding': [0.017020374536514282, -0.023255806416273117, ...],
+    #  'meta': {'model': 'ibm/slate-30m-english-rtrvr',
+    #           'truncated_input_tokens': 3}}
+    ```
+    """
+
+    def __init__(
+        self,
+        *,
+        model: str = "ibm/slate-30m-english-rtrvr",
+        api_key: Secret | None = None,
+        url: str = "https://us-south.ml.cloud.ibm.com",
```

Comment:
Like in doc embedder perhaps we should rename this to `api_base_url` for consistency with ChatGenerator
```suggestion
        api_base_url: str = "https://us-south.ml.cloud.ibm.com",
```